### DISCUSSION TITLE
What is the best approach for ArgoCD installation

### DISCUSSION THREAD

**Discussion by mohamedfazrin** [argoproj/argo-cd#11300] [Q&A]
Hi,
I have 4 Kubernetes Clusters.  I have installed ArgoCD using helm charts in each of the Kubernetes clusters. The CI/CD operations are working fine.

But, I'm looking for a new design including below:
- Install ArgoCD outside the Kubernetes Cluster using docker. 
- That single argoCD installation will manage all 4 k8s clusters.
 
I would like to know what is the best approach to follow when we have multiple clusters.

**Comment by tuatara73** [comment]
You should try using "Clusters"

**Comment by andrewhearin** [comment]
I bootstrap argo to it's own "management" k8s cluster, then register external clusters to it to be able to deploy to. I don't really see the value in having argo running on every cluster when we manage multiple clusters, having a one stop shop makes way more sense in my case and is far less much to manage.

**Comment by kostis-codefresh** [comment]
If you are asking about how to do it see https://argo-cd.readthedocs.io/en/stable/user-guide/commands/argocd_cluster_add/

If you are asking what is the best approach, there isn't a correct/wrong answer. See https://codefresh.io/blog/scaling-argo-cd-securely-in-2022/
